File: src/ui/modules/montr-core/models/data-view.ts
import { Button, DataColumn, DataPane, IDataField } from ".";

export interface DataView<TEntity> {
	id: string;
	toolbar?: Button[];
	panes?: DataPane<TEntity>[];
	panels?: DataPane<TEntity>[];
	columns?: DataColumn[];
	fields?: IDataField[];
}

File: src/Montr.Metadata/Models/DataView.cs
using System.Collections.Generic;

namespace Montr.Metadata.Models
{
	public class DataView
	{
		public string Id { get; set; }

		public ICollection<Button> Toolbar { get; set; }

		public ICollection<DataPane> Panes { get; set; }

		public ICollection<DataColumn> Columns { get; set; }

		public ICollection<FieldMetadata> Fields { get; set; }
	}
}
